### ISSUE TITLE
[Accessibility]: HeroPage icon link missing a label

### ISSUE THREAD

**Issue by kopancek** [sourcegraph/sourcegraph#35698]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34112

### Problem description

When using the HeroPage react component, links on the top of the page are represented as an image. If we use the sourcegraph logo icon there, we have 2 problems as a result:
1. The svg is focusable
2. The link does not contain any text or `aria-label`, so the screen reader cannot interpret where the link goes to or what is the purpose of the link

### Expected behavior

1. The svg itself is not focusable
2. The link contains a label or description, that is not visible by default. If the description/label was not provided, the label should contain the url the link points to

### Additional details

<img width="315" alt="Screenshot 2022-05-19 at 13 43 05" src="https://user-images.githubusercontent.com/9974711/169285566-9dcffeb0-7317-4de6-889b-5ae9b6e6b889.png">
<img width="316" alt="Screenshot 2022-05-19 at 13 43 25" src="https://user-images.githubusercontent.com/9974711/169285575-e224624c-a1c0-4627-a80f-7a15b11d2550.png">

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.
